crates/hotshot/types/src/epoch_membership.rs

```diff
@@ -228,7 +228,7 @@ where
         else {
             return self.clone().catchup(epoch).await;
         };
-        let Ok(Ok(mem)) = rx.recv_direct().await else {
+        let Ok(mem) = rx.recv_direct().await else {
```

Comment:
this looks wrong, we get mem is a Result here, the outer ok is just if the recv suceeds

Reply:
so I think it should return the result received so that wait_for_catchup() returns the error if catchup failed

Reply:
or I guess we do want to block. I'll revert this change

Reply:
I in the case of `Ok(Err(_))` we want to just return the `Err` this way the error will bubble up.

Maybe it's ok as is, we'll just loop trying to catchup to the new